lib/msf/core/payload/linux/x64/reverse_tcp_x64.rb

```diff
@@ -95,7 +95,7 @@ def asm_reverse_tcp(opts={})
 
     asm = %Q^
       mmap:
-        xor    rdi, rdi
+        xor    edi, edi
```

Comment:
I'm running through this and trying to figure out if this really does clear the upper bits or not.  Do you have a citation I could see?  If it does not clear the upper bits, we're sort of setting ourselves up again for a similar problem?

Reply:
In volume 1 chapter 3.4.1.1 of [Intel® 64 and IA-32 Architectures Software Developer’s Manual](https://cdrdv2.intel.com/v1/dl/getContent/671200).

> When in 64-bit mode, operand size determines the number of valid bits in the destination general-purpose register:
> - 32-bit operands generate a 32-bit result, zero-extended to a 64-bit result in the destination general-purpose register.

Reply:
I want you to understand how much I appreciate citing Intel's manual.  Thank you for that.